btor/src/boolector_solver.cpp

```diff
@@ -935,6 +925,30 @@ Term BoolectorSolver::apply_prim_op(PrimOp op, Term t0, Term t1, Term t2) const
           std::static_pointer_cast<BoolectorTerm>(t0);
       result = boolector_apply(btor, &args[0], 2, bt0->node);
     }
+    else if (op == Forall)
+    {
+      std::shared_ptr<BoolectorTerm> bt0 =
+          std::static_pointer_cast<BoolectorTerm>(t0);
+      std::shared_ptr<BoolectorTerm> bt1 =
+          std::static_pointer_cast<BoolectorTerm>(t1);
+      std::shared_ptr<BoolectorTerm> bt2 =
+          std::static_pointer_cast<BoolectorTerm>(t2);
+      std::vector<BoolectorNode *> params({ bt0->node, bt1->node });
+      return std::make_shared<BoolectorTerm>(
+          btor, boolector_forall(btor, &params[0], 2, bt2->node));
```

Comment:
You seem to be telling boolector to expect 2 params but then only passing in the first rather than the entire vector

Reply:
Thanks! It's actually correct, it's just a strange pattern. Since boolector is a C API they don't have vectors, so you can pass the address of the first element (which it treats as an array) and the number of elements. However, someone taught me recently that I can instead use `params.data()` which is much easier to read. I made that change in 496e4533b1a8e0fe1c844281d12f7996db19df43. Thanks for pointing this out!